utest/parsing/test_lexer.py

```diff
@@ -1816,5 +1816,199 @@ def _verify(self, data, expected, test=False):
         assert_tokens(data, expected, data_only=True)
 
 
+class TestContinue(unittest.TestCase):
+
+    def test_in_keyword(self):
+        data = '    CONTINUE'
+        expected = [(T.CONTINUE, 'CONTINUE', 3, 4),
+                    (T.EOS, '', 3, 12)]
+        self._verify(data, expected)
+
+    def test_in_test(self):
+        # This is not valid usage but that's not recognized during lexing.
+        data = '    CONTINUE'
+        expected = [(T.CONTINUE, 'CONTINUE', 3, 4),
+                    (T.EOS, '', 3, 12)]
+        self._verify(data, expected, test=False)
+
+    def test_in_if(self):
+        data = '''\
+    FOR    ${x}    IN    @{STUFF}
+        IF    True
+            CONTINUE
+        END
+    END
+'''
+        expected = [(T.FOR, 'FOR', 3, 4),
+                    (T.VARIABLE, '${x}', 3, 11),
+                    (T.FOR_SEPARATOR, 'IN', 3, 19),
+                    (T.ARGUMENT, '@{STUFF}', 3, 25),
+                    (T.EOS, '', 3, 33),
+                    (T.IF, 'IF', 4, 8),
+                    (T.ARGUMENT, 'True', 4, 14),
+                    (T.EOS, '', 4, 18),
+                    (T.CONTINUE, 'CONTINUE', 5, 12),
+                    (T.EOS, '', 5, 20),
+                    (T.END, 'END', 6, 8),
+                    (T.EOS, '', 6, 7)
+                    (T.END, 'END', 7, 4),
+                    (T.EOS, '', 6, 7)]
+        self._verify(data, expected)
+
+
+    def test_in_try(self):
+        data = '''\
+    FOR    ${x}    IN    @{STUFF}
+        TRY
+            CONTINUE
+        END
+    END
+'''
+        expected = [(T.FOR, 'FOR', 3, 4),
+                    (T.VARIABLE, '${x}', 3, 11),
+                    (T.FOR_SEPARATOR, 'IN', 3, 19),
+                    (T.ARGUMENT, '@{STUFF}', 3, 25),
+                    (T.EOS, '', 3, 33),
+                    (T.TRY, 'IF', 4, 8),
+                    (T.EOS, '', 4, 11),
+                    (T.CONTINUE, 'CONTINUE', 5, 12),
+                    (T.EOS, '', 5, 20),
+                    (T.END, 'END', 6, 8),
+                    (T.EOS, '', 6, 7)
+                    (T.END, 'END', 7, 4),
+                    (T.EOS, '', 6, 7)]
+        self._verify(data, expected)
+
+
+    def test_in_for(self):
+            data = '''\
+    FOR    ${x}    IN    @{STUFF}
+        CONTINUE
+    END
+'''
+            expected = [(T.FOR, 'FOR', 3, 4),
+                    (T.VARIABLE, '${x}', 3, 11),
+                    (T.FOR_SEPARATOR, 'IN', 3, 19),
+                    (T.ARGUMENT, '@{STUFF}', 3, 25),
+                    (T.EOS, '', 3, 33),
+                    (T.END, 'END', 7, 4),
+                    (T.EOS, '', 6, 7)]
+            self._verify(data, expected)
```

Comment:
There's no `CONTINUE` in expected. I assume this doesn't succeed yet.

Reply:
Corrected.